lib/liquid/block_body.rb

```diff
@@ -193,7 +193,7 @@ def blank?
     # Note that it is now preferred to use the `liquid` tag for this use case.
     def remove_blank_strings
       raise "remove_blank_strings only support being called on a blank block body" unless @blank
-      @nodelist.reject! { |node| node.instance_of?(String) }
+      @nodelist.reject! { |node| node.is_a?(String) }
```

Comment:
Do you actually have a use case for string subclass being used in all these derived string classes?  Can't we just coerce to `String` before tokenizing?

Reply:
Also seems like a viable solution. The author's use case is due to use of ActiveSupport::SafeBuffer in the application.

Do you think that should go in `Tokenizer#initialize`, and duplicated in `C::Tokenizer#initialize`?

Reply:
Yeah.  We can just use `source = source.to_str unless source.instance_of?(String)` for the coercion.

Reply:
Doesn't look like this is a problem for liquid-c